### DISCUSSION TITLE
Change Midi Input Channel within Setup

### DISCUSSION THREAD

**Discussion by carlitzt** [FortySevenEffects/arduino_midi_library#349] [Q&A]
Hi all,

I would like to be able to change the given Midi In Channel to a different channel, while I am in the setup routine (or maybe even in the main loop).

Example would be a midi foot controller, where I set the midi in channel with the following logic:

- power up the foot controller
- check if one special button is pressed
- if pressed, start midi in channel routine
- wait for next midi in signal and grab the midi channel
- set the midi in channel to the grabbed midi in channel

With that behavior, I can set the midi in channel by an external device.

Any recommendation on how to implement this?

Main question is, can I set the midi in channel in a void loop?

Thanks for your help

**Comment by franky47** [comment]
Yes, you can call `MIDI.setInputChannel(value)` whenever you want. Values are:
- 1 to 16 for a specific channel
- `MIDI_CHANNEL_OMNI` to listen to everything
- `MIDI_CHANNEL_OFF` to disable MIDI input entirely

If needed, you can read the current value of the input channel with `MIDI.getInputChannel()`
